### DISCUSSION TITLE
Which version of clang-formart is used by clangd in vscode?

### DISCUSSION THREAD

**Discussion by qiaomo** [clangd/clangd#1356] [Q&A] [answered]
The clangd version I use is 15.0.3, and the `AlignAfterOpenBracket: BlockIndent` option will report an error:
```
error: unknown enumerated scalar
AlignAfterOpenBracket: BlockIndent
                       ^~~~~~~~~~~
```
But when I use clang-format 14.0.0 to test, this option can work normally, so, is clangd using a lower version of clang-format？And how can I make this option work in vscode?

**Comment by HighCommander4** [comment, marked as answer]
Clangd uses the libraries of the corresponding clang-format version, e.g. 15.0.3 in this case, so this should work.

Is it possible the diagnostic is produced by a different plugin that may be using an older version? (You can check by temporarily disabling the clangd plugin.)

If not, could you share a complete `.clang-format` file for further diagnosis?

**Reply by qiaomo** [reply]
Thanks for your reply!

I found the reason. When I moved the 'clangd.path' configuration in the '[cpp]' block outside, the problem was solved.

I think the clangd vscode extension may not be able to read the config in '[cpp]' block correctly and the OS default clangd is used, which is causing the problem.

```
// invalid configuration
"[cpp]": {
    "clangd.path": "/Users/xxx/Library/Application Support/Code/User/globalStorage/llvm-vs-code-extensions.vscode-clangd/install/15.0.3/clangd_15.0.3/bin/clangd",
},
```
```
// move it out
"[cpp]": {
    ...
},
"clangd.path": "/Users/xxx/Library/Application Support/Code/User/globalStorage/llvm-vs-code-extensions.vscode-clangd/install/15.0.3/clangd_15.0.3/bin/clangd",
```
